subsys/spm/secure_services.c

```diff
@@ -54,6 +55,13 @@ int spm_secure_services_init(void)
 
 
 #ifdef CONFIG_SPM_SERVICE_READ
+
+#define FICR_BASE               NRF_FICR_S_BASE
```

Comment:
Ideally, these are generic memory defines, and could be moved into a soc-specific header, could they?

Reply:
Up to now, secure_service happens only in nRF9160. An extra header file will complicate the situation, so I would rather not recommend. 

A better solution is to rename NRF_FICR_S to NRF_FICR in zephyr\ext\hal\nordic\nrfx\mdk\nrf9160.h. FICR does not have non-secure mapping, so there is no need to explicitly distinguish between secure/non-secure. Since the physical base address is the same as other nRF5 series, using the same name can result in better code reuse.

A simpler solution is to add one more line of #define in zephyr\ext\hal\nordic\nrfx\mdk\nrf9160.h:

`#define NRF_FICR NRF_FICR_S`

This simplifies things.